ql/src/java/org/apache/hadoop/hive/ql/metadata/Hive.java

```diff
@@ -1530,7 +1530,11 @@ public void truncateTable(String dbDotTableName, Map<String, String> partSpec, L
       List<String> partNames = ((null == partSpec)
               ? null : getPartitionNames(table.getDbName(), table.getTableName(), partSpec, (short) -1));
       if (snapshot == null) {
-        getMSC().truncateTable(table.getDbName(), table.getTableName(), partNames);
+        if (table.getSnapshotRef() != null && table.getStorageHandler().isTableMetaRefSupported()) {
```

Comment:
````
String fullTableName = table.getTableName();
if (table.getSnapshotRef() != null && table.getStorageHandler().isTableMetaRefSupported())  {
 fullTableName += "." + table.getSnapshotRef();
}
````
can we move this into table.getFullTableName?

Reply:
we can do but we already have on getFullTableName
```
  public TableName getFullTableName() {
```
this will create a ``TableName`` object & return, from which we get the concatenated table name, so object creation will redundant & Java won't let us create another ``getFullTableName`` with just different return type, will any other name work?

Reply:
hm, we already have `getFullTableName()` and deprecated `getFullyQualifiedName()`
maybe let's keep in inline?